### ISSUE TITLE
npm run hot not working on windows/homestead

### ISSUE THREAD

**Issue by trogne** [JeffreyWay/laravel-mix#1931]
-   Laravel Mix Version: ^4.0.7 (`npm list --depth=0`)
-   Node Version (`node -v`): 10.13.0
-   NPM Version (`npm -v`):  6.4.1
-   OS: Windows 10

### Description:
"npm run hot" not working on default laravel/vue installation.  Laravel on homestead vm.

### Steps To Reproduce:

Install a fresh Laravel app on Homestead.
On Windows 10, run "npm install".

Then : "npm install", and "npm run hot".

Two problems :  First, public/js/app.js not created.
But main problem is that hot loading is not working.

These are NOT working : 
https://github.com/JeffreyWay/laravel-mix/blob/master/docs/hot-module-replacement.md
https://github.com/JeffreyWay/laravel-mix/issues/1147

**Comment by GENL** [comment]
I have the same issue.

**Comment by jlsjonas** [comment]
seems broken (again) since 3.0; fix for <3:

```js
Mix.listen('configReady', (webpackConfig) => {
    if (Mix.isUsing('hmr')) {
        // Remove leading '/' from entry keys
        webpackConfig.entry = Object.keys(webpackConfig.entry)
            .reduce((entries, entry) => {
                console.log('entry', entry);
                entries[entry.replace(/^\//, '')] = webpackConfig.entry[entry];
                return entries;
            }, {});

        // Remove leading '/' from ExtractTextPlugin instances
        webpackConfig.plugins.forEach((plugin) => {
            if (plugin.constructor.name === 'ExtractTextPlugin') {
                console.log('plugin', plugin.filename);
                plugin.filename = plugin.filename.replace(/^\//, '');
            }
        });
    }
});
```

**Comment by bigperson** [comment]
> long-lasting issue it seems, fix for <3.0:
> 
> ```js
> Mix.listen('configReady', (webpackConfig) => {
>     if (Mix.isUsing('hmr')) {
>         // Remove leading '/' from entry keys
>         webpackConfig.entry = Object.keys(webpackConfig.entry)
>             .reduce((entries, entry) => {
>                 console.log('entry', entry);
>                 entries[entry.replace(/^\//, '')] = webpackConfig.entry[entry];
>                 return entries;
>             }, {});
> 
>         // Remove leading '/' from ExtractTextPlugin instances
>         webpackConfig.plugins.forEach((plugin) => {
>             if (plugin.constructor.name === 'ExtractTextPlugin') {
>                 console.log('plugin', plugin.filename);
>                 plugin.filename = plugin.filename.replace(/^\//, '');
>             }
>         });
>     }
> });
> ```

I have some problem on laravel-mix v. 4.0.7 - 4.0.15, 

![image](https://user-images.githubusercontent.com/4757391/57394445-3c470600-71ce-11e9-8801-0b2e9ccc4d6c.png)

Your decision helped me. 
Windows 10 Pro. 
Node v8.11.1
